eng/source-build-pre-release.yml

```diff
@@ -0,0 +1,146 @@
+    - name: SdkVersion
+      value: $[ stageDependencies.GetBuildInfoStage.GetBuildInfoJob.outputs['ReadReleaseInfoStep.SdkVersion'] ]
+    - name: RuntimeVersion
+      value: $[ stageDependencies.GetBuildInfoStage.GetBuildInfoJob.outputs['ReadReleaseInfoStep.RuntimeVersion'] ]
+    - name: ReleaseChannel
+      value: $[ stageDependencies.GetBuildInfoStage.GetBuildInfoJob.outputs['ReadReleaseInfoStep.ReleaseChannel'] ]
+    steps:
+    - script: |
+        set -euxo pipefail
+        echo $(ReleaseChannel)
+        echo $(RuntimeVersion)
+        echo $(SdkVersion)
+        echo $(InstallerCommit)
+      name: TestVariablesStep
+      displayName: Test Variables
+    - script: |
+        set -euo pipefail
+        az config set extension.use_dynamic_install=yes_without_prompt
+
+        installer_official_runs=$(az pipelines runs list --organization $(azDoOrg) --project $(azDoProject) --pipeline-ids $(installerOfficialCiPipelineId))
+        installer_official_run_id=$(echo $installer_official_runs | jq -r '.[] | select(.sourceVersion == "$(InstallerCommit)") | .id')
+        echo "dotnet-installer-official-ci run ID: ${installer_official_run_id}"
+
+        installer_tarball_build_runs=$(az pipelines runs list --organization $(azDoOrg) --project $(azDoProject) --pipeline-ids $(installerTarabllBuildCiPipelineId))
+        installer_tarball_build_run_id=$(echo $installer_tarball_build_runs | jq -r '.[] | select(.sourceVersion == "$(InstallerCommit)") | .id')
+        echo "dotnet-installer-source-build-tarball-build run ID: ${installer_tarball_build_run_id}"
+
+        echo "##vso[task.setvariable variable=InstallerOfficialRunId]${installer_official_run_id}"
+        echo "##vso[task.setvariable variable=InstallerTarballBuildRunId]${installer_tarball_build_run_id}"
+      name: GetAssociatedPipelineRunsStep
+      displayName: Get Associated Pipeline Run IDs
+      env:
+        AZURE_DEVOPS_EXT_PAT: $(System.AccessToken)
+    - task: DownloadPipelineArtifact@2
+      name: DownloadSourceTarballStep
+      displayName: Download Source Tarball
+      inputs:
+        source: 'specific'
+        artifact: 'BlobArtifacts'
+        patterns: 'BlobArtifacts/dotnet-sdk-source-*.tar.gz'
+        project: 'internal'
+        pipeline: $(installerOfficialCiPipelineId)
+        runVersion: 'specific'
+        runId: $(InstallerOfficialRunId)
+    - script: |
+        set -euxo pipefail
+        az config set extension.use_dynamic_install=yes_without_prompt
+
+        source_tarball_upload_path="$(blobContainerUploadBaseFilePath)/$(ReleaseChannel)/$(RuntimeVersion)-$(SdkVersion)"
+        source_tarball_filename="dotnet-sdk-source-$(SdkVersion).tar.gz"
+        az storage blob upload --account-name "$(storageAccountName)" --container-name "$(blobContainerName)" --file "$(PIPELINE.WORKSPACE)/${source_tarball_filename}" --name "${source_tarball_upload_path}/${source_tarball_filename}"
```

Comment:
Does this fail gracefully if the download step fails?  My experience is the download step silently fails when patterns are used.  Is there a way to get it to fail?  If not should this step validate it exists before trying to upload?

Reply:
It actually does fail silently. Well not silently, but it doesn't have an exit code 1. I will look into this.

Reply:
I added a check that the file is downloaded before the upload, and another check that the file is listed in blob storage afterwards.